File: problems/001-anagram-detection/csharp/Brute-force.cs
using System.Linq;
using System.Collections.Generic;
class BruteForce : ISolution {
    public object Solve(Dictionary<string, object> c) {
        string s = c["s"].ToString(), t = c["t"].ToString();
        if (s.Length != t.Length) return false;
        // Obvious approach: for every character, compare its count in both strings.
        foreach (var ch in s)
            if (s.Count(x => x == ch) != t.Count(x => x == ch)) return false;
        return true;
    }
}

File: problems/001-anagram-detection/rust/brute-force.rs
use crate::json::JVal;

pub fn solve(c: &JVal) -> JVal {
    let s: Vec<char> = c.get("s").as_str().chars().collect();
    let t: Vec<char> = c.get("t").as_str().chars().collect();
    if s.len() != t.len() { return JVal::Bool(false); }
    // Obvious approach: for every character, compare its count in both strings. O(n^2).
    for &ch in &s {
        let cs = s.iter().filter(|&&x| x == ch).count();
        let ct = t.iter().filter(|&&x| x == ch).count();
        if cs != ct { return JVal::Bool(false); }
    }
    JVal::Bool(true)
}
